### ISSUE TITLE
NVDA restart process is not robust

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#17485]
Issue found while trying to fix #11644.

NVDA restart process is not robust at all; there are many unforseen corner cases.

#### Steps to reproduce:
1. Start "`nvda --disable-add`" (please note the incomplete command line flag)
2. Press `NVDA+Q`, select "Restart" and validate
3. Repeat 2 as many times as wanted
#### Actual behavior:
After steps 2 and 3, NVDA restarts again with add-ons disabled, in contradiction to what is written in the restart dialog.
#### Expected behavior:
After steps 2 and 3, NVDA should restart with add-ons re-enabled.
#### Steps to reproduce:
1. Start "`nvda --lang=it --lang=en --disable-addons --disable-addons`"
2. Press `NVDA+Q`, select "Restart" and validate
3. Press `NVDA+Q`, select "Restart" and validate
#### Actual behavior:
After step 2, NVDA is restarted with add-ons disabled and English language.
#### Expected behavior:
After steps 2 and 3, NVDA should restart with add-ons re-enabled.
#### Steps to reproduce:
1. Start "`nvda --lan=it`"
2. Open NVDA settings dialogs
#### Actual behavior:
At step 1., no command parsing error window appears.
At step 2, NVDA appears in usual Windows language.

#### Expected behavior:
At step 2. NVDA should appear in Italian, since prefix matching should apply to command line options, even if not really advertised.
Alternatively, if such option writing is not allowed, at step 1., a command line error window should appear.

### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
Installed or from source
#### NVDA version:
2024.4.1
#### Windows version:
Windows 10 22H2 (AMD64) build **19045.5131**
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, but probably no change
#### If NVDA add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested, unrelated.

**Comment by CyrilleB79** [comment]
Note: I plan to fix this along with #11644.

**Comment by seanbudd** [comment]
Can these be implemented as separate PRs? They appear to be independent issues and this helps with ease of reverting if something goes wrong

**Comment by CyrilleB79** [comment]
Due to this bug, they are not independent: I cannot implement #11644 without reworking the restart process. Thus they will not be able to be reverted independently.

Though, if you wish, I may first implement the current issue (restart process fix) and then #11644. But it will be for clarity; independent revert will not be possible.

**Comment by seanbudd** [comment]
Thanks for explaining.

To fix this particular issue we encourage disabling [prefix matching](https://docs.python.org/3/library/argparse.html#argument-abbreviations-prefix-matching). This would be an API breaking change.

**Comment by CyrilleB79** [comment]
> To fix this particular issue we encourage disabling [prefix matching](https://docs.python.org/3/library/argparse.html#argument-abbreviations-prefix-matching). This would be an API breaking change.

Happy that you suggest this; I was about to suggest it myself for simplicity and because I doubt many people use it or even know of this possibility. I did not know myself of this possibility before I started working on this issue.

It's worth noting that disabling prefix matching may also be recommended when using unknown args as we do with command line flags for add-ons.
